### ISSUE TITLE
Document custom MediaTypes handling

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#9688]
### Issue description

> We have a custom YAML payload support implemented with the standard Jackson ObjectMapper and Micronaut MediaTypeCodec approach. It's very simple, just two small classes, but worked perfectly with MN 3.x - the codec was properly picked up by DefaultHttpClient from MediaTypeCodecRegistry and YAML payloads were serialized and deserialized as expected.
> Unfortunately it doesn't work anymore with MN 4.
Apparently there was a heavy reimplementation of the http-related logic, including the DefaultHttpClient, which now leverages the newly-introduced MessageBodyHandler API (BTW **I'm a bit surprised that such an important logic was replaced by an API which is marked as experimental without an option to switch to the old one**). It is mentioned in the documentation only once, in a statement that HttpContentProcessor was superseded by MessageBodyHandler API. **Nothing about custom MediaTypes handling, not a word in breaking changes or upgrade guide**. 
Can you please advise how to proceed with this? Should we implement a custom MessageBodyHandler in addition to the two existing classes?

@yawkat 
> yes you need to implement your own, similar to https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/json-core/src/main/java/io/micronaut/json/body/JsonMessageHandler.java

**Comment by sdelamo** [comment]
@yawkat could you document this?

**Comment by sdelamo** [comment]
@yawkat @graemerocher can someone document the [`MessageBodyHandler` API](https://docs.micronaut.io/latest/api/io/micronaut/http/body/MessageBodyHandler.html) ?

**Comment by graemerocher** [comment]
Triage: the report is a valid Micronaut 4 documentation/migration gap. Custom codecs that worked through the old `MediaTypeCodec` path need a `MessageBodyHandler`/writer registered for the application media type and payload type; the guide should show the handler implementation, indexing/registration, `isWriteable` media-type matching, and both client and server wiring. The API is in Core, so the issue remains here while the user-guide/API documentation ownership is clarified; no code reproducer is needed for this documentation request.
